### ISSUE TITLE
Help users deal with datetime arithmetic

### ISSUE THREAD

**Issue by msarrel** [arrow-py/arrow#886]
## Feature Request

There is a fundamental choice when doing date time arithmetic across time changes for daylight savings time.  One option is to always treat a day as exactly 24 hours.  In that case if you add a day across a time change, the "wall clock" time of the event shifts by an hour.  The other choice is to treat days as things of flexible length, such that the "wall clock" time remains fixed when you add a day across a time change.  Both options are valid choices.

This feature request is to augment the Arrow API to better expose these choices to the user, and help them get the behavior that they desire.  This could take the form of optional arguments to various methods that would give users explicit control of what type of arithmetic is used.

For example, the default behavior of range() and shift() would stay the same.  But, if the option is set by the user, range() and shift() would translate their inputs to UTC, perform their operation, and then translate the results back to the original time zone.  I'm sure similar solutions could apply to other methods.

I suspect this is the root cause of several issues, including at least:  https://github.com/arrow-py/arrow/issues/871, https://github.com/arrow-py/arrow/issues/884 and https://github.com/arrow-py/arrow/issues/885.

For further reference see:

[PEP 431](https://www.python.org/dev/peps/pep-0431/) and specifically the [Withdrawal section](https://www.python.org/dev/peps/pep-0431/#withdrawal)

[PEP 615](https://www.python.org/dev/peps/pep-0615/)

[Semantics of timezone-aware datetime arithmetic](https://blog.ganssle.io/articles/2018/02/aware-datetime-arithmetic.html)

[A curious case of non-transitive datetime comparison](https://blog.ganssle.io/articles/2018/02/a-curious-case-datetimes.html)

cc: @areyal

**Comment by msarrel** [comment]
@systemcatch Thanks for forwarding those two blog posts.  They really helped clarify things for me.  I've adjusted the above feature request accordingly.

**Comment by b10011** [comment]
> translate their inputs to UTC, perform their operation, and then translate the results back to the original time zone

This was my initial approach, but I found another bug during daylight saving time conversions:

```python
# The UTC times below would translate to 03:00:00+03:00 and 03:00:00+02:00 Europe/Helsinki time
first_3am = arrow.get(2020, 10, 25, 0, 0, tzinfo="utc")
second_3am = arrow.get(2020, 10, 25, 1, 0, tzinfo="utc")

# Here's the bug
first_3am.to("Europe/Helsinki").timestamp
second_3am.to("Europe/Helsinki").timestamp
```

Both timestamps are the same after converting the timezone to Europe/Helsinki due to Arrow not understanding to convert second_3am to UTC+2. Instead Arrow thinks both times are in UTC+3.

@msarrel Do you think I should report this as a separate bug or is these all related to each other?

**Comment by systemcatch** [comment]
Python's datetime module made certain choices about dealing with ambiguous/imaginary times and how to handle timezones. Arrow relies heavily on dateutil and datetime for tz management and the amount of work (assuming its possible) to change this behaviour isn't worth it. So we will not be implementing the requested feature.

@b10011 what version of arrow are you using? That bug is fixed in 0.17.0.

**Comment by b10011** [comment]
> Python's datetime module made certain choices about dealing with ambiguous/imaginary times and how to handle timezones. Arrow relies heavily on dateutil and datetime for tz management and the amount of work (assuming its possible) to change this behaviour isn't worth it. So we will not be implementing the requested feature.
> 
> @b10011 what version of arrow are you using? That bug is fixed in 0.17.0.

Oh, good catch, for some reason my version was 0.15.6. I didn't consider the possibility that another package could have installed this one way earlier and I just ran "pip install arrow" when I found out about this library, which doesn't upgrade the package.

Upgraded to 0.17.0 and can confirm that I couldn't reproduce the bug. Thanks!
